src/Repository.js

```diff
@@ -43,7 +44,34 @@ export default class Repository {
     return this.lernaJson && this.lernaJson.bootstrapConfig || {};
   }
 
+  get packages() {
+    if (!this._packages) {
+      this.buildPackageGraph();
+    }
+    return this._packages;
+  }
+
+  get filteredPackages() {
+    if (!this._filteredPackages) {
+      this.buildPackageGraph();
+    }
+    return this._filteredPackages;
+  }
+
+  get packageGraph() {
+    if (!this._packageGraph) {
+      this.buildPackageGraph();
```

Comment:
Shouldn't these be called with arguments? I see `buildPackageGraph` is called in `runPreparations`, but that seems a bit like spooky action at a distance.

Perhaps `Repository` should only cache `packages`/`packageGraph` and filtering should be performed explicitly by the task that wants to filter (i.e. `Command`, as it does now).